pkg/skaffold/build/tag/tag.go

```diff
@@ -32,21 +35,39 @@ type Tagger interface {
 	GenerateTag(workingDir, imageName string) (string, error)
 }
 
+const DeprecatedImageName = "_DEPRECATED_IMAGE_NAME_"
+
 // GenerateFullyQualifiedImageName resolves the fully qualified image name for an artifact.
 // The workingDir is the root directory of the artifact with respect to the Skaffold root,
 // and imageName is the base name of the image.
 func GenerateFullyQualifiedImageName(t Tagger, workingDir, imageName string) (string, error) {
+	// Supporting the use of the deprecated {{.IMAGE_NAME}} in envTemplate
+	if v, ok := t.(*envTemplateTagger); ok {
+		tag, err := v.GenerateTag(workingDir, DeprecatedImageName)
+
+		if err != nil {
+			return "", fmt.Errorf("generating deprecated envTemplate tag: %w", err)
```

Comment:
I don't think deprecated is really relevant here — we don't actually know that they've used `{{.IMAGE_NAME}}`.
```suggestion
			return "", fmt.Errorf("generating envTemplate tag: %w", err)
```

Reply:
Good catch